tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/Logging/NLog/LogsInjection/LoggerImplWriteIntegration.cs

```diff
@@ -0,0 +1,87 @@
+// <copyright file="LoggerImplWriteIntegration.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Threading;
+using Datadog.Trace.ClrProfiler.CallTarget;
+
+namespace Datadog.Trace.ClrProfiler.AutoInstrumentation.Logging.NLog.LogsInjection
+{
+    /// <summary>
+    /// LoggerImpl.Write calltarget instrumentation
+    /// </summary>
+    [InstrumentMethod(
+        AssemblyName = "NLog",
+        TypeName = "NLog.LoggerImpl",
+        MethodName = "Write",
+        ReturnTypeName = ClrNames.Void,
+        ParameterTypeNames = new[] { ClrNames.Type, "NLog.Internal.TargetWithFilterChain", "NLog.LogEventInfo", "NLog.LogFactory" },
+        MinimumVersion = "2.1.0",
+        MaximumVersion = "4.*.*",
+        IntegrationName = "NLog")]
+    [Browsable(false)]
+    [EditorBrowsable(EditorBrowsableState.Never)]
+    public class LoggerImplWriteIntegration
+    {
+        /// <summary>
+        /// OnMethodBegin callback
+        /// </summary>
+        /// <typeparam name="TTarget">Type of the target</typeparam>
+        /// <typeparam name="TTargets">The type of the TargetWithFilterChain </typeparam>
+        /// <typeparam name="TLogEventInfo">The type of the LogEventInfo</typeparam>
+        /// <typeparam name="TLogFactory">The type of the LogFactory</typeparam>
+        /// <param name="instance">Instance value, aka `this` of the instrumented method.</param>
+        /// <param name="loggerType">The instance of the logger type</param>
+        /// <param name="targetsForLevel">The instance of the targets for the level</param>
+        /// <param name="logEvent">The logging event instance</param>
+        /// <param name="factory">The LogFactory instance</param>
+        /// <returns>Calltarget state value</returns>
+        public static CallTargetState OnMethodBegin<TTarget, TTargets, TLogEventInfo, TLogFactory>(TTarget instance, Type loggerType, TTargets targetsForLevel, TLogEventInfo logEvent, TLogFactory factory)
+        {
+            var tracer = Tracer.Instance;
+
+            if (tracer.Settings.LogsInjectionEnabled)
+            {
+                if (DiagnosticContextHelper.Cache<TTarget>.Mdlc is { } mdlc)
```

Comment:
In a followup PR, I'd like to investigate how we could add this info on the LogEvent property itself and if such an approach could remove NLog.config changes. Just a thought I had while I was here 🧠

Reply:
Yeah, that was my initial thought, but that will break consumer's config, as they explicitly state where the values come from (i.e. mdc/mdlc). Maybe we take that approach in more recent versions of NLog where the typical config is "include everything from mdc/mdlc" rather than "print mdc.dd.service" etc?